jupyter_kernel_mgmt/hl.py

```diff
@@ -0,0 +1,79 @@
+"""High-level APIs"""
+from contextlib import contextmanager
+from .client import IOLoopKernelClient, BlockingKernelClient
+from .util import run_sync
+from .discovery import KernelFinder
+
+async def start_kernel_async(name, cwd=None, launch_params=None, finder=None):
+    """Start a kernel by kernel type name, return (manager, async client)"""
+    kf = finder or KernelFinder.from_entrypoints()
+    conn_info, km = await kf.launch(name, cwd=cwd, launch_params=launch_params)
+    kc = IOLoopKernelClient(conn_info, manager=km)
+    try:
+        await kc.wait_for_ready()
+    except:
+        await kc.shutdown_or_terminate()
+        kc.close()
```

Comment:
Do we need to force the shutdown of the kernel process during exception handling?  I've found that `km.kill()` has been necessary, especially since the process has launched but the client may not be working correctly.

I'd prefer to use `km.cleanup()` there, but then that implies that implementations should make sure `cleanup()` calls `kill()` as appropriate.

Reply:
`kc.shutdown_or_terminate()` includes calling `manager.kill()` if the kernel doesn't shut itself down. Previously it could have missed this if the kernel sent a shutdown reply but then didn't shut down, but the changes in `client.py` should catch that case as well now.

Reply:
Dang, I should have checked on that.  I forgot that `KernelClient` has the manager.